src/mapping/epoch-action.ts

```diff
@@ -40,3 +40,35 @@ export function handleEpochUnstakeSuccess(
     entity.save();
   }
 }
+
+export function handleEpochCleanup(
+  data: TypedMap<string, JSONValue>,
+  receipt: near.ReceiptWithOutcome
+): void {
+  const timestamp = receipt.block.header.timestampNanosec;
+
+  const stakeAmountToSettle = BigInt.fromString(data.get('stake_amount_to_settle')!.toString());
+  const unstakeAmountToSettle = BigInt.fromString(data.get('unstake_amount_to_settle')!.toString());
+
+  // Use epochId as id
+  const id = receipt.block.header.epochId.toBase58();
+
+  const entity = EpochCleanup.load(id);
+  if (entity) {
```

Comment:
maybe we can simplify the logic by removing the entire `if (entity) {` code block and saving new `EpochCleanup` entity only if `!entity` is `true`.

Reply:
It's a defense to have us informed when unexpected situations happen.

Reply:
Do you mean catching the thrown errors in logs? 

In practice, it's hard to browse and track the logs subgraph, but I think it's OK to keep the above logic.